File: 2term.labs/lab 2.3/c++/Text.cpp
#include "Text.h"
#include <algorithm>
#include "String.h"

Text::Text(vector<String>& stringss) : strings(stringss) {};

void Text::addString(const String& string1) {
    strings.push_back(string1);
}

void Text::deleteAllText() {
    strings.clear();
}

void Text::deleteString(const String& string1) {
    auto it = std::remove_if(strings.begin(), strings.end(), [&](const String& s) {
        return s.getString() == string1.getString();
        });
    strings.erase(it, strings.end());
}

string Text::TheLargestString() {
    if (strings.empty()) return "";

    auto it = std::max_element(strings.begin(), strings.end(), [](const String& a, const String& b) {
        return a.getString().length() < b.getString().length();
        });

    return it->getString();
}
 
double Text::getPercentOfDigits() {
    int totalDigits = 0;
    int totalChars = 0;

    for (const auto& str : strings) {
        totalDigits += str.numberOfDigits();
        totalChars += str.getString().length();
    }

    if (totalChars == 0) return 0.0;
    return (100.0 * totalDigits) / totalChars;
}

int Text::getNumbersOfSymbols() {
    int total = 0;
    for (const auto& str : strings) {
        total += str.getString().length();
    }
    return total;
}

File: 2term.labs/lab 2.3/c#/Text.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClassLibrary1
{
    public class Text
    {
        private List<MyString> strings;

        public Text(List<MyString> stringss)
        {
            strings = new List<MyString>(stringss);
        }

        public void AddString(MyString string1)
        {
            strings.Add(string1);
        }

        public void DeleteString(MyString string1)
        {
            strings.RemoveAll(s => s.GetString() == string1.GetString());
        }

        public void DeleteAllText()
        {
            strings.Clear();
        }

        public string TheLargestString()
        {
            if (strings.Count == 0) return "";

            return strings.OrderByDescending(s => s.GetString().Length).First().GetString();
        }

        public double GetPercentOfDigits()
        {
            int totalDigits = 0;
            int totalChars = 0;

            foreach (var str in strings)
            {
                totalDigits += str.NumberOfDigits();
                totalChars += str.GetString().Length;
            }

            if (totalChars == 0) return 0.0;
            return 100.0 * totalDigits / totalChars;
        }

        public int GetNumbersOfSymbols()
        {
            int total = 0;
            foreach (var str in strings)
            {
                total += str.GetString().Length;
            }
            return total;
        }
    }
}